tests/unit/checkboxradio/checkboxradio_methods.js

```diff
@@ -0,0 +1,140 @@
+/*
+ * checkboxradio_methods.js
+ */
+
+
+(function($) {
+
+module( "Checkboxradio: methods" );
+	test( "Checkbox: refresh", function() {
+		var checkbox = $( "#checkbox-method-refresh" ),
+			widget;
+		expect( 2 );
+		checkbox.checkboxradio({
+			icon: true
+		});
+
+		widget = checkbox.checkboxradio( "widget" );
+		delete $(".ui-icon" )[0];
+
+		checkbox.checkboxradio( "refresh" );
+		strictEqual( widget.find( ".ui-icon" ).length, 1, "Icon is recreated on refresh" );
+		checkbox.prop( "checked", true );
+		checkbox.checkboxradio( "refresh" );
+		strictEqual( widget.hasClass( "ui-checkboxradio-checked" ), true,
+			"State updated based on checked property" );
+	});
+
+	test( "Checkbox: destroy", function(){
```

Comment:
Try to rewrite this with `domEqual()`, should simplify this test a bit and make it much more precise.